lib/rouge/lexers/haxe.rb

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Haxe < RegexLexer
+      title "Haxe"
+      desc "Haxe Cross-platform Toolkit (http://haxe.org)"
+
+      tag 'haxe'
+      aliases 'hx', 'Haxe', 'haxe'
+      filenames '*.hx'
+      mimetypes 'text/haxe', 'text/x-haxe', 'text/x-hx'
+
+      state :comments_and_whitespace do
+        rule /\s+/, Text
+        rule %r(//.*?$), Comment::Single
+        rule %r(/\*.*?\*/)m, Comment::Multiline
+      end
+
+      state :expr_start do
+        mixin :comments_and_whitespace
+
+        rule %r(/) do
+          token Str::Regex
+          goto :regex
+        end
+
+        rule /[{]/, Punctuation, :object
+
+        rule //, Text, :pop!
+      end
+
+      state :regex do
+        rule %r(/) do
+          token Str::Regex
+          goto :regex_end
+        end
+
+        rule %r([^/]\n), Error, :pop!
+
+        rule /\n/, Error, :pop!
+        rule /\[\^/, Str::Escape, :regex_group
+        rule /\[/, Str::Escape, :regex_group
+        rule /\\./, Str::Escape
+        rule %r{[(][?][:=<!]}, Str::Escape
+        rule /[{][\d,]+[}]/, Str::Escape
+        rule /[()?]/, Str::Escape
+        rule /./, Str::Regex
+      end
+
+      state :regex_end do
+        rule /[gim]+/, Str::Regex, :pop!
+        rule(//) { pop! }
+      end
+
+      state :regex_group do
+        # specially highlight / in a group to indicate that it doesn't
+        # close the regex
+        rule /\//, Str::Escape
+
+        rule %r([^/]\n) do
+          token Error
+          pop! 2
+        end
+
+        rule /\]/, Str::Escape, :pop!
+        rule /\\./, Str::Escape
+        rule /./, Str::Regex
+      end
+
+      state :bad_regex do
+        rule /[^\n]+/, Error, :pop!
+      end
+
+      def self.keywords
```

Comment:
Formatting is inconsistent. Maybe it could be enhanced either manually or with a tool?